substrate/client/network/sync/src/chain_sync.rs

```diff
@@ -2085,7 +2073,14 @@ where
 		}
 
 		self.allowed_requests.set_all();
-		Box::new(output.into_iter())
+
+		self.actions.append(&mut output);
```

Comment:
Why don't we directly push to `self.actions`?

Reply:
We need to do `self.actions.extend(self.restart())` in multiple places in this function, but we can't do that due to borrowing issues because `restart()` returns an iterator. So we accumulate all the actions until the end of the function in a separate `Vec`.

Reply:
Why isn't `restart` directly pushing to `self.actions` as well?

Reply:
Yeah, makes sense. Fixed.